terminal/pages/history.go

```diff
@@ -33,60 +41,107 @@ func historyPage(wallet walletcore.Wallet, hintTextView *primitives.TextView, tv
 	})
 
 	messageTextView := primitives.WordWrappedTextView("")
-	displayMessage := func(message string, error bool) {
+	displayMessage := func(message string, kind messageKind) {
 		// this function may be called from a goroutine, use tviewApp.QueueUpdateDraw
 		tviewApp.QueueUpdateDraw(func() {
-			body.RemoveItem(messageTextView)
+			historyPage.RemoveItem(messageTextView)
 			if message != "" {
-				if error {
+				if kind == ErrorMessage {
 					messageTextView.SetTextColor(helpers.DecredOrangeColor)
 				} else {
 					messageTextView.SetTextColor(tcell.ColorWhite)
 				}
 
 				messageTextView.SetText(message)
-				body.AddItem(messageTextView, 2, 0, false)
+				historyPage.AddItem(messageTextView, 2, 0, false)
 			}
 		})
 	}
 
-	// get total tx count early on, so as to display on history header
-	txCount, txCountErr := wallet.TransactionCount(nil)
-	totalTxCount = txCount
-
 	// page title
-	historyPageTitle := fmt.Sprintf("History (%d transactions)", txCount)
-	titleTextView := primitives.NewLeftAlignedTextView(historyPageTitle)
-	body.AddItem(titleTextView, 2, 0, false)
-
-	if txCountErr != nil {
-		displayMessage(fmt.Sprintf("Cannot load history. Get total tx count error: %s", txCountErr.Error()), true)
-		tviewApp.SetFocus(body)
-		return body
+	titleTextView := primitives.NewLeftAlignedTextView("History")
+	historyPage.AddItem(titleTextView, 2, 0, false)
+
+	// tx filter
+	var transactionCountByFilter []string
+
+	// var txCount int
+	// var txCountErr error
+	for _, filter := range walletcore.TransactionFilters {
+		if filter == "All" {
+			txCount, txCountErr := wallet.TransactionCount(nil)
+			if txCountErr != nil {
+				displayMessage(fmt.Sprintf("Cannot load history page. Error getting total transaction count: %s", txCountErr.Error()), ErrorMessage)
+				tviewApp.SetFocus(historyPage)
+				return historyPage
+			}
+
+			totalTxCount = txCount
+		}
+
+		txCount, txCountErr := wallet.TransactionCount(walletcore.BuildTransactionFilter(filter))
+		if txCountErr != nil {
+			displayMessage(fmt.Sprintf("Cannot load history page. Error getting total transaction count: %s", txCountErr.Error()), ErrorMessage)
```

Comment:
This is not an error getting TOTAL transaction count. Replace with
```suggestion
			displayMessage(fmt.Sprintf("Cannot load history page. Error getting transaction count for filter %s: %s", filter, txCountErr.Error()), ErrorMessage)
```

Also, reading this line of code makes it obvious that the variable `ErrorMessage` is poorly named. Rename to `MessageKindError`